runtime/ops/command.rs

```diff
@@ -0,0 +1,332 @@
+) -> Result<Command, AnyError> {
+  let args = command_args.cmd;
+  super::check_unstable(state, "Deno.Command");
+  state.borrow_mut::<Permissions>().run.check(&args[0])?;
+
+  let mut command = Command::new(&args[0]);
+  command.args(&args[1..]);
+
+  if let Some(cwd) = command_args.cwd {
+    command.current_dir(cwd);
+  }
+
+  if command_args.clear_env {
+    command.env_clear();
+  }
+  command.envs(command_args.env);
+
+  command.stdin(subprocess_stdio_map(&command_args.stdin)?);
+  command.stdout(subprocess_stdio_map(&command_args.stdout)?);
+  command.stderr(subprocess_stdio_map(&command_args.stderr)?);
+
+  // We want to kill child when it's closed
+  command.kill_on_drop(true);
+
+  Ok(command)
+}
+
+#[derive(Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct CommandStatus {
+  success: bool,
+  code: i32,
+  signal: Option<i32>,
+}
+
+impl From<std::process::ExitStatus> for CommandStatus {
+  fn from(status: ExitStatus) -> Self {
+    let code = status.code();
+    #[cfg(unix)]
+    let signal = status.signal();
+    #[cfg(not(unix))]
+    let signal = None;
+
+    if let Some(signal) = signal {
+      CommandStatus {
+        success: false,
+        code: 128 + signal,
+        signal: Some(signal),
+      }
+    } else {
+      let code = code.expect("Should have either an exit code or a signal.");
+
+      CommandStatus {
+        success: code == 0,
+        code,
+        signal: None,
+      }
+    }
+  }
+}
+
+async fn op_command_status(
+  state: Rc<RefCell<OpState>>,
+  command_args: CommandArgs,
+  _: (),
+) -> Result<CommandStatus, AnyError> {
+  let mut command = create_command(&mut state.borrow_mut(), command_args)?;
+  let status = command.status().await?;
+
+  Ok(status.into())
+}
+
+#[derive(Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct CommandOutput {
+  status: CommandStatus,
+  stdout: Option<ZeroCopyBuf>,
+  stderr: Option<ZeroCopyBuf>,
+}
+
+async fn op_command_output(
+  state: Rc<RefCell<OpState>>,
+  command_args: CommandArgs,
+  _: (),
+) -> Result<CommandOutput, AnyError> {
+  let stdout = command_args.stdout.clone();
+  let stderr = command_args.stderr.clone();
+  let mut command = create_command(&mut state.borrow_mut(), command_args)?;
+  let output = command.output().await?;
+
+  Ok(CommandOutput {
+    status: output.status.into(),
+    stdout: if &stdout == "piped" {
+      Some(output.stdout.into())
+    } else {
+      None
+    },
+    stderr: if &stderr == "piped" {
+      Some(output.stderr.into())
+    } else {
+      None
+    },
+  })
+}
+
+#[derive(Serialize)]
+#[serde(rename_all = "camelCase")]
+struct Child {
+  rid: ResourceId,
+  pid: u32,
+  stdin_rid: Option<ResourceId>,
+  stdout_rid: Option<ResourceId>,
+  stderr_rid: Option<ResourceId>,
+}
+
+fn op_command_spawn(
+  state: &mut OpState,
+  command_args: CommandArgs,
+  _: (),
+) -> Result<Child, AnyError> {
+  let mut command = create_command(state, command_args)?;
+  let mut child = command.spawn()?;
```

Comment:
There's a trade-off here, while this certainly is simpler than mapping commands 1:1 it also means that if `create_command` errors out then that'll happen here rather than in `new Command(...)` which is what I would expect as the user of such a constructor.

Reply:
Agreed, will do this, especially as i need this for some future plans

Reply:
Done!